Improve helm release.md (#4939)
grafana/mimir: operations/helm/charts/mimir-distributed/RELEASE.md

1. Prepare changelog.

   - Create a PR, whose target is `main`, that updates the [Helm chart changelog](https://github.com/grafana/mimir/blob/main/operations/helm/charts/mimir-distributed/CHANGELOG.md), and move any `## main / unreleased` items under this release’s version.

   > **Note:** If there are any deprecated features that should be removed in this release, then verify that they have been removed, and move their deprecation notices into the section for this release.

   - Have the PR reviewed by a maintainer.

   - Merge the PR upon approval.

1. Create a release branch.

   - Create (if the branch is not created yet), switch to and push a branch starting from the commit created by the prepare changelog PR and name it `mimir-distributed-release-x.y`.

     For example, `mimir-distributed-release-4.5` for any `4.5.x` release.

   - Push the branch to origin without any commit added.

   - Once the branch is pushed, all changes to `mimir-distributed-release-x.y` branch must be done through PR.

1. Create a branch from release branch if it hasn't been created yet, to update Mimir/GEM image and helm chart version .

   For example `user/update-mimir-distributed-release-x.y`.

> 1. Update versions and links in the `user/update-mimir-distributed-release-x.y` branch.

Reviewer: ```suggestion
1. Prepare the release on the `user/update-mimir-distributed-release-x.y` branch.
```

This makes sense if we extract the version setting into its own step. WDYT?

   - Set the image versions in [values.yaml](https://github.com/grafana/mimir/blob/main/operations/helm/charts/mimir-distributed/values.yaml), as needed:

     - `image.tag` (Mimir)
     - `enterprise.image.tag` (GEM)

       > **Note:** Unlike the Mimir image tags, GEM image tags start with `v`. For example, `v2.6.0` instead of `2.6.0`.

     - `smoke_test.image.tag` (Smoke test; usually the same as Mimir)
     - `continuous_test.image.tag` (Continuous test; usually the same as Mimir)

>    - Set the `version` field, in the [Chart.yaml](https://github.com/grafana/mimir/blob/main/operations/helm/charts/mimir-distributed/Chart.yaml) file, to the desired release candidate version.

Reviewer: Let's extract this in a new step (will be step 6). So the preparations can be done without risk of releasing anything.
Let's call the step "Trigger the release"

Reviewer: we agreed to keep it here

     For example, `4.5.0-rc.0`.

     > **Note:** Once this change is merged to `mimir-distributed-x.y` branch, it will trigger the release process GitHub Action. Unless you want to release final release, make sure to append the correct rc version.

   - Set the `appVersion` field, in the [Chart.yaml](https://github.com/grafana/mimir/blob/main/operations/helm/charts/mimir-distributed/Chart.yaml) file, to the version of Mimir that the Helm chart deploys.

     For example, `2.6.0`.

   - Create or update the release notes in [release notes](https://github.com/grafana/mimir/tree/mimir-distributed-release-4.2/docs/sources/helm-charts/mimir-distributed/release-notes).

>      The release notes should refer to the correct Mimir and GEM versions and their specific documentation version.

Reviewer: Add a `Note` that this step can be done in a separate PR and delayed until final release if it takes a long time.

   - Go back to step 5. Update versions and links in the user/update-mimir-distributed-release-x.y branch to update the next release candidate.

The [release process](https://github.com/grafana/mimir/blob/main/.github/workflows/helm-release.yaml) checks and creates a Git tag formatted as `mimir-distributed-<version>`, for example `mimir-distributed-4.5.0`, on the merge commit created when the PR is merged. To prevent releasing the same version with different content, the release process fails if the tag already exists. The release is published in the [Grafana helm-charts](https://grafana.github.io/helm-charts/) Helm repository.

## Release process for a final release

1. Determine the Helm chart version number.

   [The Chart.yaml file](https://helm.sh/docs/topics/charts/#the-chartyaml-file) requires semantic versioning:

   - The final version has the version `x.y.z`, for example `3.1.0`.
   - Normally we will proceed the same `x.y.z` version value from the release candidate step.

1. Create a branch from release branch to update Mimir/GEM image and helm chart version.

   For example `user/update-mimir-distributed-release-x.y-final`.

> 1. Update versions in the `user/update-mimir-distributed-release-x.y-final` branch.

Reviewer: Above this add a step: finalize release notes if not done already

   - Set the `version` field, in the [Chart.yaml](https://github.com/grafana/mimir/blob/main/operations/helm/charts/mimir-distributed/Chart.yaml) file, to the desired final release version.

     For example, `4.5.0`.

     > **Note:** Once this change is merged to `mimir-distributed-x.y` branch, it will trigger the release process GitHub Action.

   - There shouldn't be anymore update needed in release notes and documentation because that has been done in the release candidate step above.

> 1. Open PR to release branch

Reviewer: Add step above to run `make doc`  (as the chart version is put into the README.md)
